src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1693,6 +1744,122 @@ private void PutCertificate(ClusterOperationContext context, string type, Blitta
             }
         }
 
+        private void BulkPutReplicationCertificate(ClusterOperationContext context, string type, BlittableJsonReaderObject cmd, long index, ServerStore serverStore)
+        {
+            var command = (BulkRegisterReplicationHubAccessCommand)CommandBase.CreateFrom(cmd);
+            try
+            {
+                var certs = context.Transaction.InnerTransaction.OpenTable(ReplicationCertificatesSchema, ReplicationCertificatesSlice);
+
+                foreach (RegisterReplicationHubAccessCommand inner in command.Commands)
+                {
+                    PutRegisterReplicationHubAccessInternal(context, inner, certs);
+                }
+            }
+            finally
+            {
+                NotifyDatabaseAboutChanged(context, command.Database, index, type, DatabasesLandlord.ClusterDatabaseChangeType.ValueChanged, command);
+            }
+        }
+        
+        private void PutReplicationCertificate(ClusterOperationContext context, string type, BlittableJsonReaderObject cmd, long index, ServerStore serverStore)
+        {
+            var command = (RegisterReplicationHubAccessCommand)CommandBase.CreateFrom(cmd);
+            try
+            {
+                var certs = context.Transaction.InnerTransaction.OpenTable(ReplicationCertificatesSchema, ReplicationCertificatesSlice);
+
+                PutRegisterReplicationHubAccessInternal(context, command, certs);
+            }
+            finally
+            {
+                NotifyDatabaseAboutChanged(context, command.Database, index, type, DatabasesLandlord.ClusterDatabaseChangeType.ValueChanged, command);
+            }
+        }
+
+
+        private unsafe void PutRegisterReplicationHubAccessInternal(ClusterOperationContext context, RegisterReplicationHubAccessCommand command, Table certs)
+        {
+            using (Slice.From(context.Allocator, (command.Database + "/" + command.HubDefinitionName + "/" + command.CertThumbprint).ToLowerInvariant(), out var keySlice))
+            using (Slice.From(context.Allocator, (command.Database + "/" + command.HubDefinitionName + "/" + command.CertPublicKeyHash).ToLowerInvariant(),
+                out var publicKeySlice))
+            using (var obj = context.ReadObject(new DynamicJsonValue
```

Comment:
Shouldn't we use command.ToJson?

Reply:
No, they are not the same object, we are only saving some of the propertioes